src/code/fault_ootn64.c

```diff
@@ -0,0 +1,835 @@
+                        f32 comboTimeS = OS_CYCLES_TO_USEC(osGetTime() - comboStartTime) / 1000000.0f;
+
+                        osSyncPrintf("入力時間 %f 秒\n", comboTimeS);
+                        if (comboTimeS <= 50.0f) {
+                            x = 0xB;
+                        } else {
+                            x = 0;
+                        }
+                    } else {
+                        x = 0;
+                    }
+                    break;
+
+                default:
+                    break;
+            }
+        }
+    }
+}
+
+extern s32 D_80105A90_unknown; // Arena_failcnt
+
+void func_800AF0E0(void) {
+    s32 i;
+
+    Fault_DrawRecBlack(22, 16, 276, 208);
+    Fault_DrawText(40, 30, "SegmentBaseAddress");
+
+    for (i = 0; i < ARRAY_COUNT(gSegments); i++) {
+        Fault_DrawText(40, 40 + 8 * i, "%2d:%08x", i, gSegments[i]);
+    }
+    Fault_DrawText(40, 180, "Arena_failcnt = %d", D_80105A90_unknown);
+}
+
+void Fault_DrawMemDumpContents(const char* title, void* memory, u32 arg2) {
+    s32 x;
+    s32 y;
+    u32* ptr = (u32*)((u32)memory & ~3);
+
+    Fault_DrawRecBlack(22, 16, 276, 208);
+    Fault_DrawText(36, 18, "%s %08x", title != NULL ? title : "PrintDump", ptr);
+
+    if ((ptr >= (u32*)0x80000000) && (ptr <= (u32*)0x80400000)) {
+        for (y = 28; y != 226; y += 9) {
+            Fault_DrawText(28, y, "%06x", ptr);
+            for (x = 82; x < 290; x += 52) {
+                Fault_DrawText(x, y, "%08x", *(ptr++));
+            }
+        }
+    }
+}
+
+void Fault_DrawMemDumpPC(OSThread* arg0) {
+    u32* pc = (u32*)arg0->context.pc;
+
+    if ((pc >= (u32*)0x80000020) && (pc <= (u32*)0x80400000)) {
+        Fault_DrawMemDumpContents("PC Trace", pc - 0x20, 0);
+    }
+}
+
+void Fault_DrawMemDumpSP(OSThread* arg0) {
+    u32* sp = (u32*)(u32)arg0->context.sp;
+
+    if ((sp >= (u32*)0x80000020) && (sp <= (u32*)0x80400000)) {
+        Fault_DrawMemDumpContents("Stack Trace", sp, 0);
+    }
+}
+
+void func_800AF3DC(void) {
+    s32 i;
+
+    Fault_DrawRecBlack(22, 16, 276, 208);
+    Fault_DrawText(36, 18, "ROM DEBUG");
+
+    for (i = 0; i < ARRAY_COUNT(B_80122570); i++) {
+        Fault_DrawText(((i % 4) * 52) + 82, ((i / 4) * 9) + 28, "%08x", B_80122570[i]);
+    }
+}
+
+void Fault_ResumeThread(OSThread* thread) {
+    thread->context.cause = 0;
+    thread->context.fpcsr = 0;
+    thread->context.pc += 4;
+    *(u32*)thread->context.pc = 0x0000000D;
+    osWritebackDCache((void*)thread->context.pc, 4);
+    osInvalICache((void*)thread->context.pc, 4);
+    osStartThread(thread);
+}
+
+void func_800AF558(void) {
+    osViBlack(false);
+    if (osViGetCurrentFramebuffer() >= (void*)0x80100000) {
+        gFaultMgr.framebuffer = (s32)osViGetCurrentFramebuffer();
+    } else {
+        gFaultMgr.framebuffer = (s32)(void*)(PHYS_TO_K0(osMemSize) - sizeof(u16[SCREEN_HEIGHT][SCREEN_WIDTH]));
+        osViSwapBuffer((void*)gFaultMgr.framebuffer);
```

Comment:
I think gFaultMgr.framebuffer should be `void*`, it would be consistent with the GC version too (and there it's called `fb`, might be worth using the same name)

Reply:
https://github.com/zeldaret/oot/pull/2046#discussion_r1721062893

Reply:
comment about `fb` still stands, and if this gets changed to `u16*` then it would be nice to sync that with the GC version too. Maybe just add a TODO for this though

Reply:
(resolved)